src/C++/SocketMonitor.h

```diff
@@ -51,6 +52,18 @@ class SocketMonitor
 public:
   class Strategy;
 
+  class SocketOperator
+  {
+  public:
+    virtual ~SocketOperator(){}
+    virtual bool fdIsSet(int i, fd_set& set) {
+      return FD_ISSET(i, &set);
+    }
+    virtual int selectExecution(int set_size, fd_set* readset, fd_set* writeset, fd_set* exceptset, timeval* timeout) {
+      return select( set_size, readset, writeset, exceptset, timeout );
+    }
+
+  };
```

Comment:
Wrapper class for lower level socket calls.  Unit Test coverage needed this to be added to allow for unit test 100% coverage